src/Infrastructure/EventHandling/DomainEventHandler/CreateAddressChangeHandler.php

```diff
@@ -55,6 +59,29 @@ public function onMembershipCreated( MembershipCreatedEvent $event ): void {
 		$this->persistIfNeeded( $addressChangeBuilder->build() );
 	}
 
+	public function onDonorUpdated( DonorUpdatedEvent $event ): void {
+		if ( !$this->donorIsAnonymous( $event->getPreviousDonor() ) || $this->donorIsAnonymous( $event->getNewDonor() ) ) {
+			return;
+		}
+
+		$addressChangeBuilder = AddressChangeBuilder::create();
+		$addressChangeBuilder->forDonation( $event->getDonationId() );
+
+		if ( $event->getNewDonor()->isPrivatePerson() ) {
+			$addressChangeBuilder->forPerson();
+		} elseif ( $event->getNewDonor()->isCompany() ) {
+			$addressChangeBuilder->forCompany();
+		}
+		$this->persistIfNeeded( $addressChangeBuilder->build() );
+	}
+
+	private function donorIsAnonymous( Donor $donor ) {
```

Comment:
Sorry to nitpick here, but email donors are not fully anonymous, so  `donorHasNoAddress` or `donorHasAddress` would be better names.